# VNC_scRNAseq
Analysis of scRNA-seq of the Drosophila melanogaster adult ventral nerve cord from females and males

Fastq sequencing data were processed using Drop-seq_tools v2.1.0, and was implemented with the '10x_parallelprocessing.sh', '10x_preprocessing.sh', and '10x_postprocessing.sh'. Resulting digital expression matrices were analysed with Seurat v2.3.4 using the 'R_preprocessing.R' script. The 'all_analysis.R' script contains information regarding the subsequent analysis for each figure of the manuscript. 'myDotPlot.R' is a modified version of Seurat v2.3.4's 'DotPlot' function to accept 3 colours when plotting the scaled expression. 'number_of_class_in_cells.R' is a function of count the number of genes in a given list per cell. 'scRNA_bulkFlyAtlas2_comparison.R' is a function to plot the correlation between pseudo-bulk signal from single cell data to FlyAtlas2 bulk RNA-seq data (assumes that you have downloaded the FlyAtlas2 data and extracted it from the SQL database).
